### PR TITLE
Upmerge 16.05.2024

### PR DESCRIPTION
**Pull request by rlubos**
Pull upmerged sdk-zephyr and sdk-mcuboot repositories.

test-sdk-nrf: sdk-nrf-pr-15351
test_ble: DRGN-22017_upmerge-16-05-2024
test_rs: KRKNWK-19014-upmerge-20240516
test-sdk-sidewalk: sdk-nrf-pr-15351

### CODE REVIEW

**Comment by MarekPieta** [comment]
Spotted a BLE VS commands related issue while testing nRF Desktop. Aligned here: https://github.com/rlubos/sdk-nrf/pull/20
Waiting for reviews from codeowners

**Review by SeppoTakalo — APPROVED** [review]

File: applications/asset_tracker_v2/src/cloud/cloud_codec/lwm2m/lwm2m_codec.c [hunk 1]
```diff
@@ -28,7 +28,8 @@ static cloud_codec_evt_handler_t module_evt_handler;
 
 /* Function that is called whenever the configuration object is written to. */
 static int config_update_cb(uint16_t obj_inst_id, uint16_t res_id, uint16_t res_inst_id,
-			    uint8_t *data, uint16_t data_len, bool last_block, size_t total_size)
+			    uint8_t *data, uint16_t data_len, bool last_block,
+			    size_t total_size, size_t offset)
```
Comment (balaji-nordic) [inline applications/asset_tracker_v2/src/cloud/cloud_codec/lwm2m/lwm2m_codec.c:32]: We need an `ARG_UNUSED(offset);` statement after line 44.

File: applications/asset_tracker_v2/tests/lwm2m_codec_helpers/src/lwm2m_codec_helpers_test.c [hunk 2]
```diff
@@ -54,7 +54,8 @@ static int date_time_now_stub(int64_t *unix_time_ms, int no_of_calls)
  * test_codec_helpers_setup_configuration_object_handler.
  */
 static int callback(uint16_t obj_inst_id, uint16_t res_id, uint16_t res_inst_id,
-		    uint8_t *data, uint16_t data_len, bool last_block, size_t total_size)
+		    uint8_t *data, uint16_t data_len, bool last_block,
+		    size_t total_size, size_t offset)
 {
 	ARG_UNUSED(obj_inst_id);
 	ARG_UNUSED(res_id);
```
Comment (balaji-nordic) [inline applications/asset_tracker_v2/tests/lwm2m_codec_helpers/src/lwm2m_codec_helpers_test.c:61]: Add `ARG_UNUSED(offset);`

Comment (rlubos) [reply applications/asset_tracker_v2/tests/lwm2m_codec_helpers/src/lwm2m_codec_helpers_test.c:61]: Fixed

**Review by kkasperczyk-no — APPROVED** [review]

**Review by jayteemo — APPROVED** [review]

**Review by plskeggs — APPROVED** [review]

**Review by MarkusLassila — APPROVED** [review]

**Review by jfischer-no — APPROVED** [review]

**Review by jfischer-no — APPROVED** [review]

File: subsys/net/lib/softap_wifi_provision/softap_wifi_provision.c [hunk 3]
```diff
@@ -798,22 +798,24 @@ static void reset_entry(void *o)
 static const struct smf_state state[] = {
 	[STATE_INIT]	      = SMF_CREATE_STATE(init_entry,
 						 init_run,
-						 NULL),
+						 NULL, NULl, NULl),
```
Comment (glarsennordic) [inline subsys/net/lib/softap_wifi_provision/softap_wifi_provision.c:801]: Argh. Sorry. I was reviewing as you merged, just noticed this.

It seems these NULLs have a lower-case L

**Review by glarsennordic — COMMENTED** [review]
Since I just found what looks like a breaking change, and this was just merged, I am submitting my review before it is complete

File: samples/wifi/sta/sample.yaml [hunk 4]
```diff
@@ -29,6 +29,7 @@ tests:
     extra_configs:
       - CONFIG_NET_TX_STACK_SIZE=3700
       - CONFIG_NET_RX_STACK_SIZE=3700
+      - CONFIG_WPA_SUPP_THREAD_STACK_SIZE=6144
```
[the remark this answers is not in the document]
Comment (rlubos) [reply samples/wifi/sta/sample.yaml:32]: @krish2718 Should it? I guess it'd make sense?

File: subsys/net/lib/softap_wifi_provision/softap_wifi_provision.c (hunk 3 above)
Comment (rlubos) [reply subsys/net/lib/softap_wifi_provision/softap_wifi_provision.c:801]: How in the world CI didn't catch this... Will fix

File: samples/wifi/sta/sample.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (krish2718) [reply samples/wifi/sta/sample.yaml:32]: Ideally yes, but this was mainly a hot fix for `nRF52840`, I will check if this is applicable to other boards as well and update `modules/hostap` if its only `nRF52840` then will need to add a board specific overlay. Thanks @glarsennordic .

**Pull request merged**